src/main/java/gift/user/controller/UserController.java

```diff
@@ -0,0 +1,45 @@
+package gift.user.controller;
+
+import gift.user.dto.LoginRequestDto;
+import gift.user.dto.LoginResponseDto;
+import gift.user.dto.RegisterRequestDto;
+import gift.user.dto.RegisterResponseDto;
+import gift.user.service.UserService;
+import jakarta.validation.Valid;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/users")
+public class UserController {
+
+  private final UserService userService;
+
+  public UserController(UserService userService) {
+    this.userService = userService;
+  }
+
+  @PostMapping("/register")
+  public ResponseEntity<RegisterResponseDto> registerUser(
+      @Valid @RequestBody RegisterRequestDto registerRequestDto) {
+
+    return new ResponseEntity<RegisterResponseDto>(userService.registerUser(registerRequestDto),
+        HttpStatus.CREATED);
+
+  }
```

Comment:
```suggestion
  @PostMapping("/register")
  public ResponseEntity<RegisterResponseDto> registerUser(
      @Valid @RequestBody RegisterRequestDto registerRequestDto) {
    var registerResponse = userService.registerUser(registerRequestDto);
    return new ResponseEntity<RegisterResponseDto>(registerResponse,
        HttpStatus.CREATED);
  }
```
가독성을 위해 분리하면 좋을 것 같아요 :)

Reply:
분리하는게 훨씬 깔끔하네요 수정하겠습니다!
멘토님은 var을 사용하시는군요..!

Reply:
저도 구체적인 타입을 쓰는 걸 좋아하는데 code suggestion 때문에 var로 썼어요 ㅎㅎ